### PR TITLE
Fix enhanced adept weapons' stats

### PR DESCRIPTION
**Pull request by robojumper**
Fairly self-explanatory.

### CODE REVIEW

**Comment by robojumper** [comment]
To elaborate on the conundrum, the example weapon from the Discord has 1 point of reload, stability, handling more in-game than it does in DIM with this patch, because adept weapons get +3 to secondary MW stats. It has the [`3249407402`](https://data.destinysets.com/i/InventoryItem:3249407402) intrinsic plugged, which gives +2 to secondary MW stats, and DIM uses that and thus ends up with one point too little.

A small survey on data.destinysets
```js
$InventoryItem.filter((i) => i.plug && i.investmentStats?.some((stat) => stat.value === 3 && stat.isConditionallyActive)).show()
```
shows that the only plugs that conditionally give +3 are the T10 masterworks, but enhanced adept raid weapons have this replaced with a dummy masterwork plug. So the game must have some interesting behavior here that the defs don't capture.

**Comment by robojumper** [comment]
Alternatively, maybe I should level up an enhanced weapon to level 20 to see if it gets both the +2 and the +3 to secondary stats...

**Comment by robojumper** [comment]
okay, wild. This is an adept Rufus's Fury (corkscrew, ricochet) at level 17 (enhanced handling intrinsic)

grafik

and this at level 20

grafik

so enhancing gives +3 to secondary MW stats, and leveling up to level 20 gives an additional +1, for a total of +4.

**Comment by robojumper** [comment]
Okay, I implemented something that calculates these stats correctly and displays them nicely but it awkwardly cheats a bit in the middle of the code where we build all items' stats...

**Review by nev-r — APPROVED** [review]
this is showing the right numbers for different levels of a RoN adept trace
(plain, masterworked, enhanced, enhanced + lvl20)

File: src/app/inventory/store/stats.ts [hunk 1]
```diff
@@ -426,15 +439,40 @@ function applyPlugsToStats(
...
+  createdItem: DimItem,
+  plug: PluggableInventoryItemDefinition,
+  stat: DestinyItemInvestmentStatDefinition
+) {
+  if (
+    stat.isConditionallyActive &&
+    enhancedIntrinsics.has(plug.hash) &&
+    adeptWeaponHashes.includes(createdItem.hash)
+  ) {
+    return stat.value * ((createdItem.craftedInfo?.level ?? 0) >= 20 ? 2 : 1.5);
```
Comment (nev-r) [inline src/app/inventory/store/stats.ts:464]: this feels like a really complicated way, based on a coincidence of numbers, to effectively say
`+ (level >= 20 ? 1 : 2)` or whatever the right numbers would be.
any reason you're going by ratio/multiplication here? does it apply to some/any other case?

**Review by bhollis — APPROVED** [review]
Wild

**Pull request merged**